server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java

```diff
@@ -172,13 +184,14 @@ private MappingLookup(
             }
         }
 
+        PassThroughObjectMapper.checkForInvalidPriorities(passThroughMappers);
```

Comment:
I'm not sure if this is called when creating or updating an index template as it doesn't affect a concrete index, yet. In any case, I think it would be prudent to add a test case for an index template with two passthrough field types that don't specify a priority to see if this gets caught.

If this indeed turns out not to be the appropriate place for the validation, an alternative may be to validate this in the root object mapper. Yet another approach could be to allow duplicate priorities and then fall back to lexicographical ordering if the priority is the same.

Reply:
So templates are validated in `MetaIndexTemplateServervice#validateTemplate()` line 1812.
The merge method at some point creates a new `DocumentMapper` instance and then the MappingLookup gets  created.

Reply:
Nice, thanks for double-checking!